code/iaas/api-logic/src/main/java/io/cattle/platform/iaas/api/cluster/ClusterRemoveHostActionHandler.java

```diff
@@ -0,0 +1,62 @@
+package io.cattle.platform.iaas.api.cluster;
+
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import io.cattle.platform.api.action.ActionHandler;
+import io.cattle.platform.core.constants.ClusterConstants;
+import io.cattle.platform.core.dao.ClusterHostMapDao;
+import io.cattle.platform.core.model.ClusterHostMap;
+import io.cattle.platform.core.model.Host;
+import io.cattle.platform.object.ObjectManager;
+import io.cattle.platform.object.process.ObjectProcessManager;
+import io.github.ibuildthecloud.gdapi.exception.ValidationErrorException;
+import io.github.ibuildthecloud.gdapi.request.ApiRequest;
+import io.github.ibuildthecloud.gdapi.validation.ValidationErrorCodes;
+
+public class ClusterRemoveHostActionHandler implements ActionHandler {
+
+    ObjectManager objectManager;
+    ObjectProcessManager processManager;
+    ClusterHostMapDao clusterHostMapDao;
+
+    @Override
+    public String getName() {
+        return "host.removehost";
+    }
+
+    @Override
+    public Object perform(String name, Object obj, ApiRequest request) {
+        if (!objectManager.isKind(obj, ClusterConstants.KIND)) {
+            return null;
+        }
+        Host cluster = (Host)obj;
+
+        Long hostToRemoveId = (Long)((Map)request.getRequestObject()).get(ClusterConstants.ADD_REMOVE_HOST_PARAM);
```

Comment:
A slightly safer and type safe way to do this would be

```java
DataAccessor.fromMap(request.getRequestObject())
   .withKey(ClusterConstants.ADD_REMOVE_HOST_PARAM)
   .as(Long.class);
```